lib/board.rb

```diff
@@ -30,4 +29,35 @@ def cells
   def valid_coordinate?(coordinate)
     @cells.key?(coordinate)
   end
+
+  def valid_placement?(ship, coordinates)
+    if (ship.length == coordinates.length) &&
+       (coordinates.all? { |coordinate| valid_coordinate?(coordinate) }) &&
+        !coordinates.empty?
+        consecutive_coordinates?(coordinates)
+    else
+      false
+    end
+  end
+
+  def consecutive_coordinates?(coordinates)
+    letters = coordinates.map { |coordinate| coordinate[0] }
+    numbers = coordinates.map { |coordinate| coordinate[1].to_i }
+
+    if !consecutive_letters?(letters) && consecutive_numbers?(numbers)
+      true
+    elsif consecutive_letters?(letters) && !consecutive_numbers?(numbers)
+      true
+    else 
+      false
+    end
+  end
+
+  def consecutive_numbers?(numbers)
+    numbers.each_cons(2).all? { |first, second| second - first == 1 }
+  end
+
+  def consecutive_letters?(letters)
+    letters.each_cons(2).all? { |first, second| second.ord - first.ord == 1 }
```

Comment:
.ord is confusing to me could you help me better understand what the method does and how it pertains to the project and why we need it.

Reply:
All string characters have what is called an ASCII code (also know as a Unicode code). This code/value is numerical. And in the case of alphabetical characters (A-Z or a-z) their ASCII codes are consecutive!
This means we can use "A" and "B" in comparisons that we couldn't normally use in strings.
For instance:
puts "A".ord        This outputs 65
puts "B".ord.      This outputs 66
puts "A".ord < "B".ord.      This outputs true since B is further in the alphabet and has a higher ASCII code

Reply:
Also remember that upper and lower case characters are not the same thing! They have different ASCII codes.
Fun fact: If you want to change a lower case letter to an uppercase one, just add 32 to its ordinal value!
                It also works in reverse, just subtract 32 to create a lower case letter!

Reply:
Thank you for the clarification!